cldfbench_nominalperson_cldf.py

```diff
@@ -20,63 +20,69 @@ def cmd_download(self, args):
+        for param in ['apc_order', 'constituent_order']:
+            args.writer.objects['ParameterTable'].append(dict(
+                ID=param,
+                Name=param.replace('_', ' '),
+            ))
+
         # Load data from grammarchecks
-        
-        lang_id=0
+        lids = set()
         val_id=0
         for row in self.raw_dir.read_csv('grammarchecks.csv',dicts=True):
+            if row['constituent_order_wals']:
+                assert row['constituent_order_wals'] == wals_83A[row['walscode']], '{}: {} vs. {}'.format(row['walscode'], row['constituent_order_wals'], wals_83A[row['walscode']])
             if row['include']=='y':
+                assert row['glottocode'] in glangs
+                lids.add(row['glottocode'])
                 args.writer.objects['LanguageTable'].append({
                     'ID': row['glottocode'],
                     'Name': row['lang_name'] ,
                     'Macroarea': row['area_glottolog'],
                     'Latitude': row['lat'],
                     'Longitude': row['lon'],
                     'Glottocode': row['glottocode'],
-                    'ISO639P3code': row['iso639_3'],
+                    'ISOcodes': sorted(set(re.findall(r'[a-z]{3}', row['iso639_3']))),
                     'Walscode': row['walscode']
                 })
-                args.writer.objects['ValueTable'].append({
-                    'ID': val_id,
-                    'Language_ID': row['glottocode'],
-                    'Parameter_ID': 'APC Order',
-                    'Value': row['apc_order']
-                })
-                val_id+=1
-                args.writer.objects['ValueTable'].append({
-                    'ID': val_id,
-                    'Language_ID': row['glottocode'],
-                    'Parameter_ID': 'Constituent Order',
-                    'Value': row['constituent_order']
-                })
-                val_id+=1
-            
-            
- #       for lid, rows in itertools.groupby(
- #           sorted(
- #               self.raw_dir.read_csv('examples.csv', dicts=True),
- #               key=lambda row: (row['Language_ID'], int(row['ID']))),
- #           lambda row: row['Language_ID']
- #       ):
- #           rows = list(rows)
- #           args.writer.objects['LanguageTable'].append(dict(
- #               ID=lid,
- #               Name=rows[0]['Language_Name'],
- #           ))
- #           for row in rows:
-            for row in self.raw_dir.read_csv('examples.csv',dicts=True):
+                for param in params:
+                    args.writer.objects['ValueTable'].append({
+                        'ID': val_id,
+                        'Language_ID': row['glottocode'],
+                        'Parameter_ID': param,
+                        'Value': row[param]
+                    })
+                    val_id+=1
+
+        for row in self.raw_dir.read_csv('examples.csv',dicts=True):
+            if row['Language_ID'] in lids:
```

Comment:
No we can check if we are dealing with a language that was added to LanguageTable or not.